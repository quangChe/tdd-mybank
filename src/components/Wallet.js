import React from 'react';
import { connect } from 'react-redux';
import { balance } from '../actions/my-balance';

export class Wallet extends React.Component {
  state = {
    amount: 0,
  }

  changeAmount = (e) => {
    this.setState({ amount: parseInt(e.target.value, 10) });
  }

  deposit = () => {
    this.props.deposit(this.state.amount);
  }

  render() {
    const { balance } = this.props;
    
    return (
      <div>
        <h3 className='balance'>Wallet balance: { balance }</h3>
        <br/>
        <input 
          onChange={this.changeAmount}
          className='wallet-input'/>
        <button
          className='deposit-btn'
          onClick={this.deposit}>Deposit</button>
      </div>
    )
  }
}

const mapStateToProps = (state) => ({
  balance: state
})

export default connect(mapStateToProps, null)(Wallet);