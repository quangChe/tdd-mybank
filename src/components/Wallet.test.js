import React from 'react';
import { shallow } from 'enzyme';
import { Wallet } from './Wallet';

describe('Wallet', () => {
  const mockDepositFn = jest.fn();
  const props = {
    balance: 20,
    deposit: mockDepositFn,
  };
  const wallet = shallow(<Wallet {...props} />);

  it('renders properly', () => {
    expect(wallet).toMatchSnapshot();
  })

  it('correctly displays balance from props', () => {
    expect(wallet.find('.balance').text()).toEqual('Wallet balance: 20')
  })

  it('contains an input to enter deposit or withdraw amount', () => {
    expect(wallet.find('.wallet-input').exists()).toBe(true);
  })

  describe('when the user types into Wallet input', () => {
    const INPUT_VALUE = '1000';

    beforeEach(() => 
      wallet.find('.wallet-input').simulate('change', { target: { value: INPUT_VALUE }})
    )

    it('changes the `amount` state of the Wallet', () => {
      expect(wallet.state().amount).toBe(parseInt(INPUT_VALUE, 10));
    })

    describe('and the user wants to make a deposit', () => {
      beforeEach(() => 
        wallet.find('.deposit-btn').simulate('click')
      )

      it('dispatches the `deposit()` it receives from the props with local `amount` state', () => {
        expect(mockDepositFn).toHaveBeenCalledWith(parseInt(INPUT_VALUE, 10));
      })
    })
  })
})