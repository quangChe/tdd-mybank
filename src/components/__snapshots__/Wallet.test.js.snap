// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Wallet renders properly 1`] = `
<div>
  <h3
    className="balance"
  >
    Wallet balance: 
    20
  </h3>
  <br />
  <input
    className="wallet-input"
    onChange={[Function]}
    type="number"
  />
  <button
    className="deposit-btn"
    onClick={[Function]}
  >
    Deposit
  </button>
  <button
    className="withdraw-btn"
    onClick={[Function]}
  >
    Withdraw
  </button>
</div>
`;
